service/test/BatchServerTest.cpp

```diff
@@ -242,23 +237,11 @@ TEST_F(BatchServerTest, run_batch_read)
 
 TEST_F(BatchServerTest, run_batch_read_empty)
 {
-    SetUp_only_controls();
-
     InSequence sequence;
 
     int idx = 0;
     std::vector<double> result = {};
 
-    // {}
-    for (const auto &request : m_signal_config) {
-        EXPECT_CALL(*m_pio_ptr, push_signal(request.name, request.domain,
-                                            request.domain_idx))
-            .WillOnce(Return(idx))
-            .RetiresOnSaturation();
-        ++idx;
-    }
```

Comment:
I notice that we avoid pushing the signals by not having that for loop at all, since our `m_signal_config` is still initialized with the proper values. And the empty vector is passed into the constructor instead.
```cpp
m_signal_config = {geopm_request_s {1, 0, "CPU_FREQUENCY"},
                       geopm_request_s {2, 1, "TEMPERATURE"}};
```